Integrations/Tanium-Rest/Tanium-Rest.py

```diff
@@ -0,0 +1,914 @@
+
+    def parse_sensor_parameters(self, parameters):
+        sensors = parameters.split(';')
+        parameter_conditions = []
+
+        for sensor in sensors:
+            sensor_name = sensor.split('{')[0]
+            tmp_item = {'sensor': sensor_name, 'parameters': []}
+
+            parameters_txt = sensor.split('{')[1][:-1]
+            params = parameters_txt.split(',')
+            for param in params:
+                tmp_item['parameters'].append({
+                    'key': '||' + param.split('=')[0] + '||',
+                    'value': param.split('=')[1]
+                })
+            parameter_conditions.append(tmp_item)
+
+        return parameter_conditions
+
+    def parse_action_parameters(self, parameters):
+        parameters = parameters.split(';')
+        parameter_conditions = []
+        for param in parameters:
+            parameter_conditions.append({
+                'key': param.split('=')[0],
+                'value': param.split('=')[1]})
+
+        return parameter_conditions
+
+    def add_parameters_to_question(self, question_response, parameters):
+        if not parameters:
+            return question_response
+
+        for item in question_response.get('selects'):
+            sensor = item.get('sensor').get('name')
+
+            for parameter in parameters:
+                if parameter['sensor'] == sensor:
+                    item['sensor']['parameters'] = parameter['parameters']
+                    item['sensor']['source_hash'] = item['sensor']['hash']
+                    del item['sensor']['hash']
+
+        return question_response
+
+    def parse_question(self, text, parameters):
+        parameters_condition = []  # type: ignore
+
+        if parameters:
+            try:
+                parameters_condition = self.parse_sensor_parameters(parameters)
+            except Exception:
+                return_error('Failed to parse question parameters.')
+
+        res = self.do_request('POST', 'parse_question', {'text': text}).get('data')[0]
+
+        res = self.add_parameters_to_question(res, parameters_condition)
+        return res
+
+    def create_question(self, question_body):
+        res = self.do_request('POST', 'questions', question_body)
+        return res.get('data').get('id'), res
+
+    def parse_question_results(self, result):
+        results_sets = result.get('data').get('result_sets')[0]
+        if results_sets.get('estimated_total') != results_sets.get('mr_tested'):
+            return None
+        if results_sets.get('row_count') == 0:
+            return []
+
+        rows = []
+        columns = []
+        for column in results_sets.get('columns'):
+            columns.append(column.get('name').replace(' ', ''))
+
+        for row in results_sets.get('rows'):
+            i = 0
+            tmp_row = {}
+            for item in row.get('data'):
+                tmp_row[columns[i]] = item[0].get('text')
+                i += 1
+            rows.append(tmp_row)
```

Comment:
use zip to go over the columns and row values at once instead of using the i variable.
```
tmp_row = {}
for item, column in zip(row.get('data',[]), columns):
    tmp_row[column] = item
```